doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java

```diff
@@ -1905,7 +1905,9 @@ private void inlineSemantics( SinkEventAttributes attributes, String semantic,
     {
         if ( attributes.containsAttribute( SinkEventAttributes.SEMANTICS, semantic ) )
         {
-            writeStartTag( tag );
+            SinkEventAttributes attributesNoSemantics = ( SinkEventAttributes ) attributes.copyAttributes();
+            attributesNoSemantics.removeAttribute( SinkEventAttributes.SEMANTICS );
```

Comment:
Why is this necessary?

Reply:
*inline* Sink events may have attributes that we want to leverage (notably the *Class* attribute).

Example: *inline* Sink event with `Semantics = "code"` and `Class = "language-java"` will be output as `<code class="language-java">`.

Note: We don't want to output `<code semantics="code" class="language-java">` and that's why we need to remove the *SEMANTICS* attribute.

Admittedly, the *inlineSemantics()* function is not the most elegant piece of code but I didn't want to reshuffle the XHTML Sink too much, as it's used in many different places.